### ISSUE TITLE
Version conflicts

### ISSUE THREAD

**Issue by thodorisGeorgiou** [ilastik/nature_methods_multicut_pipeline#4]
Hey,

I am trying to run the "run_mc_2d.sh" under Ubuntu 18.04, but I get some issues.
If I try to run the scripts by just extracting the tarball I get the following output:

##########################################################################
##########################################################################
##########################################################################
##########################################################################

That, even though I have succesfully run the "install-cplex-shared-libs.sh" script.
If I uncomment the try / catch I get the following import error:

ImportError: libilocplex.so: cannot open shared object file: No such file or directory

Although libilocplex.so exists in the lib/ directory.
I tried adding the lib directory to pythons path but didn't work.

I tried building within my own miniconda2 and anaconda2 environments and get the following conflicts (same for both anaconda2 and miniconda2):

command:  conda create --name multicut-software --channel ilastik multicut_pipeline:

Collecting package metadata (current_repodata.json): done
Solving environment: failed with repodata from current_repodata.json, will retry with next repodata source.
Collecting package metadata (repodata.json): done
Solving environment: | 
Found conflicts! Looking for incompatible packages.
This can take several minutes.  Press CTRL-C to abort.
failed                               

UnsatisfiableError: The following specifications were found to be incompatible with each other:

Package opengm-thorsten conflicts for:
multicut_pipeline -> opengm-thorsten==2.3.6.post1=np19py27_12_g059e159_with_cplex
Package wsdt-mcp conflicts for:
multicut_pipeline -> wsdt-mcp==0.1=py27_1_g764505d
Package python conflicts for:
multicut_pipeline -> python=2.7
Package graph conflicts for:
multicut_pipeline -> graph==1.6=np19py27_2_g764505d
Package futures conflicts for:
multicut_pipeline -> futures
Package scikit-learn conflicts for:
multicut_pipeline -> scikit-learn==0.16.1
Package vigra-thorsten conflicts for:
multicut_pipeline -> vigra-thorsten==1.11=np19py27_3_g575e7eb
Package h5py conflicts for:
multicut_pipeline -> h5py==2.5.0

Do you have any idea what might be going wrong?

**Comment by constantinpape** [comment]
> That, even though I have succesfully run the "install-cplex-shared-libs.sh" script.
> If I uncomment the try / catch I get the following import error:
> 
> ImportError: libilocplex.so: cannot open shared object file: No such file or directory

Which cplex version do you have? Did you run the `install-cplex-shared-libs.sh` script like this?
```
$ ./install-cplex-shared-libs.sh /path/to/cplex-dir /path/to/multicut-binaries
```

> Do you have any idea what might be going wrong?

That's weird, I just tried and I can install it with
```
conda create -c ilastik -n mc-nm multicut_pipeline
```
I am also on Ubuntu 18.04.
Which conda version are you using? I have 4.6.14.
Is there anything in your `.condarc` that might cause issues?
Also, note that you need to export the `CPLEX_ROOT_DIR` environment variable before installing the conda env to set it up properly.

As a side note, the cplex dependency makes it pretty hard to install this software correctly. Since the publication, we have switched to using approximate solvers that don't rely on cplex and usually yield results of similar quality.
If you are interested, here's a small example on how we use multicut segmentation currently:
https://github.com/constantinpape/elf/blob/master/example/segmentation/multicut.ipynb
Feel free to open an issue in that repository if your interested to run this example and encounter any issues.

**Comment by thodorisGeorgiou** [comment]
>  Which cplex version do you have?

When I import in Python 2.7 it says "12.9.0.0"

> Did you run the install-cplex-shared-libs.sh script like this?

Initially yes, but I get this error:

`Usage: ./install-cplex-shared-libs.sh <path-to-CPLEX_Studio1251>`

If I run the above command I get the following output:

>++ g++ -fpic -shared -Wl,-whole-archive /home/thodoris/Documents/deepLearn/ssTEM/linuxSources/CPLEX//cplex/lib/x86-64_linux/static_pic/libcplex.a -Wl,-no-whole-archive -o /home/thodoris/Documents/deepLearn/ssTEM/linuxSources/multicut-software/lib/libcplex.so
++ g++ -fpic -shared -Wl,-whole-archive /home/thodoris/Documents/deepLearn/ssTEM/linuxSources/CPLEX//concert/lib/x86-64_linux/static_pic/libconcert.a -Wl,-no-whole-archive -o /home/thodoris/Documents/deepLearn/ssTEM/linuxSources/multicut-software/lib/libconcert.so
++ g++ -fpic -shared -Wl,-whole-archive /home/thodoris/Documents/deepLearn/ssTEM/linuxSources/CPLEX//cplex/lib/x86-64_linux/static_pic/libilocplex.a -Wl,-no-whole-archive -o /home/thodoris/Documents/deepLearn/ssTEM/linuxSources/multicut-software/lib/libilocplex.so
++ set +x
CPLEX shared libraries installed to: /home/thodoris/Documents/deepLearn/ssTEM/linuxSources/multicut-software/lib
DONE.

> Which conda version are you using?

4.7.12

>Is there anything in your .condarc that might cause issues?

Its empty (I don't use anaconda, just installed it for this project)

>Also, note that you need to export the CPLEX_ROOT_DIR environment variable before installing the conda env to set it up properly.

Just tried this and got the same errors


>If you are interested, here's a small example on how we use multicut segmentation currently:
https://github.com/constantinpape/elf/blob/master/example/segmentation/multicut.ipynb
Feel free to open an issue in that repository if your interested to run this example and encounter any issues.

If this doesn't work I will give it a try. But since I want to make comparisons to other papers as well that use your method, I want to try and have as few differences in the setup as possible.

**Comment by constantinpape** [comment]
> > Which conda version are you using?
> 
> 4.7.12

Ok, I upgraded to conda 4.7.12 now and I run into the same dependency issues as you are.
Could you switch to conda 4.6.14 and try again?
You can download older versions of the installer here:
https://repo.anaconda.com/miniconda/

Note that this seems to be an issue with dependency resolution in newer conda versions, so I think this would need to be fixed on the conda side.

**Comment by thodorisGeorgiou** [comment]
I installed conda 4.6.14 and run the commands:

export CPLEX_ROOT_DIR=/apth/to/CPLEX
conda create --name multicut-software --channel ilastik multicut_pipeline

I got the following errors:

ERROR conda.core.link:_execute_post_link_actions(658): An error occurred while installing package 'ilastik::cplex-shared-0.1-9'.
LinkError: post-link script failed for package ilastik::cplex-shared-0.1-9
running your command again with `-v` will provide additional information
location of failed script: /home/thodoris/miniconda2/envs/multicut-software/bin/.cplex-shared-post-link.sh
==> script messages <==
<None>

Attempting to roll back.

failed
ERROR conda.core.link:_execute(568): An error occurred while installing package 'ilastik::cplex-shared-0.1-9'.
LinkError: post-link script failed for package ilastik::cplex-shared-0.1-9
running your command again with `-v` will provide additional information
location of failed script: /home/thodoris/miniconda2/envs/multicut-software/bin/.cplex-shared-post-link.sh
==> script messages <==
<None>

Attempting to roll back.

Rolling back transaction: done

CondaMultiError: LinkError: post-link script failed for package ilastik::cplex-shared-0.1-9
running your command again with `-v` will provide additional information
location of failed script: /home/thodoris/miniconda2/envs/multicut-software/bin/.cplex-shared-post-link.sh



Nonetheless, the script "run_mc_2d.sh" seems to run and finish properly.
Thanks you for your time!
